arangod/Aql/Variable.cpp

```diff
@@ -43,30 +43,33 @@ char const* const Variable::NAME_CURRENT = "$CURRENT";
 
 /// @brief create the variable
 Variable::Variable(std::string name, VariableId id, bool isDataFromCollection)
-    : id(id), 
-      name(std::move(name)), 
-      value(nullptr), 
+    : id(id),
+      name(std::move(name)),
+      value(nullptr),
       isDataFromCollection(isDataFromCollection) {}
 
 Variable::Variable(arangodb::velocypack::Slice const& slice)
-    : Variable(arangodb::basics::VelocyPackHelper::checkAndGetStringValue(slice, "name"),
-               arangodb::basics::VelocyPackHelper::checkAndGetNumericValue<VariableId>(slice, "id"),
-               arangodb::basics::VelocyPackHelper::getBooleanValue(slice, "isDataFromCollection", false)) {}
+    : id(arangodb::basics::VelocyPackHelper::checkAndGetNumericValue<VariableId>(slice, "id")),
+      name(arangodb::basics::VelocyPackHelper::checkAndGetStringValue(slice, "name")),
+      value(),
+      isDataFromCollection(arangodb::basics::VelocyPackHelper::getBooleanValue(slice, "isDataFromCollection", false)),
+      hasConstValue(arangodb::basics::VelocyPackHelper::getBooleanValue(slice, "hasConstValue", false)),
+      constantValue(slice.get("constantValue")) {}
```

Comment:
I also considered that, but if we later want to introduce virtual registers it might make sense to have a separate type field. So instead I would suggest to replace the `hasConstValue` with an enum field `type`.

Reply:
Done